src/app/components/control-bar/control-bar.component.ts

```diff
@@ -165,24 +163,25 @@ export class ControlBarComponent implements OnInit {
     this.projectsService.projects.subscribe((projects) => {
       this.projects = projects;
 
-      if (this.projects.length) {
-        let lastProj;
-        // try {
-        //   //restores view to the last visited project from local storage
-        //   lastProj = JSON.parse(window.localStorage.getItem('lastProj'));
-        //   // console.log(lastProj);
-        // } catch (error) {
-        //   lastProj = this.projectsService.setActiveProject(this.projects[0]);
-        // }
-
-        lastProj = this.projectsService.setActiveProject(this.projects[0]);
-
-        // If lastProj is null, then there is no project saved, or can be found, default to the first project in the list
-        if (lastProj == 'none' || lastProj == null) {
-          lastProj = this.projects[0];
-        }
+      // restores view to the last visited project from local storage
+      let lastProject = null;
+      try {
+        lastProject = JSON.parse(window.localStorage.getItem('lastProj'));
+      } catch (error) {
+        // possible that lastProj item is null and not json
+        lastProject = null;
+      }
 
-        this.projectsService.setActiveProject(lastProj);
+      if (projects.length) {
+        const selectedLastProject = lastProject ? this.projects.find((prj) => prj.id === lastProject.id) : null;
+        if (selectedLastProject) {
+          this.projectsService.setActiveProject(selectedLastProject);
+        } else {
+          // default to the first project in the list
+          this.projectsService.setActiveProject(this.projects[0]);
+        }
+      } else {
+        this.projectsService.setActiveProject(null);
       }
```

Comment:
```suggestion
      }
```
This part seems to revert `lastProj` in the `localStorage` to `null` every time we load the page.
Since the initial value (before we get the response from `getProjects()`) of `projectsService.projects` is `[]`, it will always hit the else clause. 
It works as intended after the change.

Reply:
ah gotcha.  👍  good catch. 

i'm gonna recheck the scenario where a user delete the last map project locally before merging

Reply:
all good 💯